src/entities/swap/swaps/v2/auraBalSwaps/joinSwap.ts

```diff
@@ -0,0 +1,133 @@
+import {
+    decodeAbiParameters,
+    decodeFunctionResult,
+    encodeFunctionData,
+    zeroAddress,
+    Address,
+    Hex,
+} from 'viem';
+import { TokenAmount } from '../../../../tokenAmount';
+import { BALANCER_RELAYER } from '@/utils';
+import { balancerRelayerAbi } from '@/abi';
+import { Relayer } from '@/entities/relayer';
+import { auraBalToken, balWethAddress, auraBAL } from './constants';
+import { getJoinData } from './joinPool';
+import { getSwapData } from './swap';
+import { AuraBalSwapQueryOutput } from './auraBalSwaps';
+import { Token } from '@/entities/token';
+import { AuraBalSwapQueryInput } from './parseInputs';
+
+// token[join]8020BPT[swap]auraBAL
+export async function queryJoinSwap(
```

Comment:
It took me a while to understand the scope of these files:
- exitPool
- joinPool
- swap
- joinSwap
- swapExit

I remember these lower level files were under `actions` on the old SDK and I believe that helps navigate things. What do you think about reorganizing to something like this?
- joinSwap
- swapExit
- actions
   - exitPool
   - joinPool
   - swap

Reply:
I'll leave it as is for now, maybe if we go back and refactor to be more generalised this would be worthwhile.